pkg/agent/endpoints/sds/handler.go

```diff
@@ -284,34 +284,49 @@ func (h *Handler) buildResponse(versionInfo string, req *api_v2.DiscoveryRequest
 	// build a convenient set of names for lookups
 	names := make(map[string]bool)
 	for _, name := range req.ResourceNames {
-		names[name] = true
+		if name != "" {
+			names[name] = true
+		}
 	}
 
 	// TODO: verify the type url
-	_, defaultBundleNameBool := names[h.c.SDSDefaultBundleName]
-
-	if upd.Bundle != nil && (len(names) == 0 || defaultBundleNameBool || names[upd.Bundle.TrustDomainID()]) {
-		validationContext, err := buildValidationContext(upd.Bundle, defaultBundleNameBool, h.c.SDSDefaultBundleName)
-		if err != nil {
-			return nil, err
+	if upd.Bundle != nil {
+		switch {
+		case len(names) == 0 || names[upd.Bundle.TrustDomainID()]:
+			validationContext, err := buildValidationContext(upd.Bundle, "")
+			if err != nil {
+				return nil, err
+			}
+			resp.Resources = append(resp.Resources, validationContext)
+		case names[h.c.SDSDefaultBundleName]:
+			validationContext, err := buildValidationContext(upd.Bundle, h.c.SDSDefaultBundleName)
+			if err != nil {
+				return nil, err
+			}
+			resp.Resources = append(resp.Resources, validationContext)
 		}
-		resp.Resources = append(resp.Resources, validationContext)
 	}
 
 	for _, federatedBundle := range upd.FederatedBundles {
 		if len(names) == 0 || names[federatedBundle.TrustDomainID()] {
-			validationContext, err := buildValidationContext(federatedBundle, false, "")
+			validationContext, err := buildValidationContext(federatedBundle, "")
 			if err != nil {
 				return nil, err
 			}
 			resp.Resources = append(resp.Resources, validationContext)
 		}
 	}
 
-	_, defaultSVIDNameBool := names[h.c.SDSDefaultSVIDName]
 	for _, identity := range upd.Identities {
-		if len(names) == 0 || defaultSVIDNameBool || names[identity.Entry.SpiffeId] {
-			tlsCertificate, err := buildTLSCertificate(identity, defaultSVIDNameBool, h.c.SDSDefaultSVIDName)
+		switch {
+		case len(names) == 0 || names[identity.Entry.SpiffeId]:
+			tlsCertificate, err := buildTLSCertificate(identity, "")
+			if err != nil {
+				return nil, err
+			}
+			resp.Resources = append(resp.Resources, tlsCertificate)
+		case names[h.c.SDSDefaultSVIDName]:
```

Comment:
Ah, i missed one thing. The default "SVID" name should only apply to the first identity (i.e. the "default" identity).

Assuming the index `i` is added to the for loop rang above:
```suggestion
		case i == 0 && names[h.c.SDSDefaultSVIDName]:
```

Reply:
👍